vali_objects/position.py

```diff
@@ -519,14 +528,27 @@ def _clamp_and_validate_leverage(self, order) -> bool:
         is_first_order = len(self.orders) == 0
         proposed_leverage = self.net_leverage + order.leverage
         min_position_leverage, max_position_leverage = leverage_utils.get_position_leverage_bounds(self.trade_pair, order.processed_ms)
+
+        current_adjusted_leverage = abs(self.net_leverage) * self.trade_pair.leverage_multiplier
+        proposed_portfolio_leverage = (net_portfolio_leverage - current_adjusted_leverage +
+                                       (abs(proposed_leverage) * self.trade_pair.leverage_multiplier))
+        max_portfolio_leverage = leverage_utils.get_portfolio_leverage_cap(order.processed_ms)
+
         # we only need to worry about clamping if the sign of the position leverage remains the same i.e. position does not flip and close
         if is_first_order or self.net_leverage * proposed_leverage > 0:
-            if abs(proposed_leverage) > max_position_leverage:
-                if is_first_order or abs(proposed_leverage) >= abs(self.net_leverage):
-                    order.leverage = max(0.0, max_position_leverage - abs(self.net_leverage))
+            if (abs(proposed_leverage) > max_position_leverage
+                    or proposed_portfolio_leverage > max_portfolio_leverage):
+                if (is_first_order
+                        or abs(proposed_leverage) >= abs(self.net_leverage)
+                        or proposed_portfolio_leverage >= net_portfolio_leverage):
+                    order.leverage = max(0.0,
```

Comment:
this section is a little wild - if the proposed portfolio leverage is greater than the limit then we take up to the limit?

Reply:
yes, we take up to the lower of the two limits between position leverage and portfolio leverage